.github/workflows/ci.yml

```diff
@@ -0,0 +1,25 @@
+# This workflow will install Python dependencies, run tests and lint with a variety of Python versions
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: detect-secrets-ci
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+
+jobs:
+  main:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+        python: ['3.6', '3.7', '3.8', '3.9']
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python }}
+    - run: python -mpip install --upgrade setuptools pip tox virtualenv
+    - run: tox -e py
```

Comment:
My thought process behind this is if we just run `tox` - the runner will attempt to run tests for all py versions in tox. However we cannot ensure that the runner will have our py versions in question. Running `tox -e py` runs the installed version in the runner which we specify in the python-version matrix. This does work and does run pre-commit. 

If you have any other ideas - please let me know.

Reply:
Nice. That makes a lot of sense.

You should record that type of rationale as a comment (perhaps over the run command itself) so that you can capture the context and design notes for future developers.

Reply:
Oh, FWIW `tox` should [skip python versions that don't exist on the host](https://github.com/Yelp/detect-secrets/blob/master/tox.ini#L5). But this way is probably a little bit more efficient.